ResearchKit/Consent/ORKVisualConsentStepViewController.m

```diff
@@ -391,12 +415,23 @@ - (void)finishTransitioningAnimator:(ORKVisualConsentTransitionAnimator *)animat
     }
 }
 
+- (void)observedScrollViewDidScroll:(UIScrollView *)scrollView {
+    CGRect animationViewFrame = _animationView.frame;
+    CGPoint scrollViewBoundsOrigin = scrollView.bounds.origin;
+    animationViewFrame.origin = (CGPoint){-scrollViewBoundsOrigin.x, -scrollViewBoundsOrigin.y};
+    _animationView.frame = animationViewFrame;
+}
+
 - (void)showViewController:(ORKConsentSceneViewController *)viewController forward:(BOOL)forward animated:(BOOL)animated {
 
     if (! viewController) {
         return;
     }
     
+    // Stop old observer and start new one
+    _scrollViewObserver = [[ORKScrollViewObserver alloc] initWithTargetView:viewController.scrollView responder:self];
```

Comment:
Moreover, for robustness, the callback should also check it's the right scrollview, i.e. that it's the scroll view of the current observer (if any)